handler_test.go

```diff
@@ -127,3 +129,29 @@ func TestStreamRequestRecorder(t *testing.T) {
 		assert.Equal(t, requests, gotRequests)
 	})
 }
+
+func TestIntervalWaiter(t *testing.T) {
+	t.Run("initial call must not wait", func(t *testing.T) {
+		waiter := newIntervalWaiter(time.Millisecond * 100)
+		now := time.Now()
+		waiter.wait(context.Background())
+		assert.True(t, time.Now().Sub(now) < time.Millisecond)
```

Comment:
assert.Less is better and add a comment:
`assert.Less(t, time.Now().Sub(now), duration, "takes more than an interval for initial wait to finish")`

Reply:
Tried both `Less`, but it does not work well with `Time.duration` (might require type cast to int64)
```
 --- FAIL: TestIntervalWaiter/initial_call_must_not_wait (0.00s)
	panic: interface conversion: interface {} is time.Duration, not int64
```

Reply:
your call then. Duration is an int64 underneath anyway. It seem to be fine to cast it to int64 and use Less to compare.